source/utils/ConfigManager.h

```diff
@@ -26,12 +26,21 @@
 class CreatureDefinition;
 class Weapon;
 class SpawnCondition;
+class CreatureMood;
 
 //! \brief This class is used to manage global configuration such as network configuration, global creature stats, ...
 //! It should NOT be used to load level specific stuff. For that, there if GameMap.
 class ConfigManager : public Ogre::Singleton<ConfigManager>
 {
 public:
+    enum CreatureMoodLevel
+    {
+        Happy,
+        Neutral,
+        Upset,
+        Angry,
+        Furious
+    };
```

Comment:
I've had a look at enum class. From what I've seen, it allows forward declaration (which is really good) but they should be declared outside the class (that makes sense since the the most important reason to declare them in the class was to gather enum values).

I would also say that we should add in the coding good practices to use enum classes for enums (and start replacing the unscoped enumerations currently in the code)